fundamentals/middleware/problem-details-service/Program.cs

```diff
@@ -186,4 +183,70 @@ static bool HasPath(HttpContext context)
         _ => false
     };
 }
+
+#elif API_CONTROLLER
+using Microsoft.AspNetCore.Http.Features;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.Services.AddProblemDetails(options =>
+    options.CustomizeProblemDetails = (context) =>
+    {
+
+        var mathErrorFeature = context.HttpContext.Features
+                                                   .GetRequiredFeature<MathErrorFeature>();
+        if (HasPath(context.HttpContext))
+        {
+            (string Detail, string Type) details = mathErrorFeature.MathError switch
+            {
+                MathErrorType.DivisionByZeroError =>
+                ("Divison by zero is not defined.",
+                                           "https://wikipedia.org/wiki/Division_by_zero"),
+                _ => ("Negative or complex numbers are not valid input.",
+                                             "https://wikipedia.org/wiki/Square_root")
+            };
+
+            context.ProblemDetails.Type = details.Type;
+            context.ProblemDetails.Title = "Wrong Input";
+            context.ProblemDetails.Detail = details.Detail;
+        }
+    }
+    );
+
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+// Middleware to handle writing problem details to the response
+app.Use(async (context, next) =>
+{
+    context.Features.Set(new MathErrorFeature());
+    await next(context);
+});
+
+app.UseStatusCodePages();
+
+app.UseAuthorization();
+
+app.MapControllers();
+
+app.Run();
+
+// Check if error message is defined for selected paths.
+static bool HasPath(HttpContext context)
+{
+    return true;
+}
+
```

Comment:
@sammychinedu2ky this is going to get super messy.

Reply:
@sammychinedu2ky  I added an API controller. Should I ask `@brunolins16` if there's a cleaner solution? Even with minimal API's, that's messy.

Reply:
Yh I think that would be great. Would love to get his input and learn as well.